src/settings/MappingProfiles/detailsSections/utils.js

```diff
@@ -286,8 +286,23 @@ export const getMappingQueryFromValue = valueFromDetails => {
   return valueFromDetails?.substring(0, mappingEndPosition === -1 ? valueFromDetails.length : mappingEndPosition);
 };
 
-export const setFieldValue = ({ path, setReferenceTables, value, isDisabled }) => {
-  const reference = isDisabled ? '' : value;
+export const clearFieldValue = ({ paths, setReferenceTables, value = '', isDisabled }) => {
+  if (isDisabled) {
+    paths.forEach(path => setReferenceTables(path, value));
+  }
```

Comment:
This function shouldn't check whether the field is disabled or not, it should be checked in the place where you call this function (in useEffect e.g.). Also, this function shouldn't accept `value` as arg, it should just clear the field value so set an empty string, or `null`, because if I pass some value to the function then it won't actually clear the field, but set some value to it.